src/test/scala/model/EnemyModelTest.scala

```diff
@@ -1,38 +1,90 @@
 package model
 
-import model.entities.{Enemy, Entity, PeaShooter, TroopState, Plant, WorldSpace, Zombie}
+
+import model.entities.{Bullet, FastZombie, PeaBullet, PeaShooter, Troop, Troops, Zombie, WarriorZombie}
 import org.scalatest.*
 import org.scalatest.flatspec.*
 import org.scalatest.matchers.*
-import WorldSpace.{LanesLength, given}
 import model.entities.TroopState.*
 
 import scala.concurrent.duration.FiniteDuration
 
 
 class EnemyModelTest extends AnyFlatSpec with should.Matchers:
 
-  "A zombie" should "enter attacking state if interests list is not empty" in {
-    val turret: Plant = PeaShooter((1, 0))
-    val zombie: Enemy =  Zombie((1, LanesLength / 2))
-    zombie.state shouldBe Moving
-    zombie.update(FiniteDuration(16, "milliseconds"), List(turret)).state shouldBe Attacking
-  }
-  "A zombie" should "not be in attacking state if interests list is empty" in {
-    val zombie: Enemy = Zombie((2, LanesLength / 2))
-    zombie.state shouldBe Moving
-    zombie.update(FiniteDuration(16, "milliseconds"), List.empty).state should not be Attacking
-  }
-  "A enemy" should "filter the interesting entities" in {
-    val basicZombie: Enemy = Zombie((1, LanesLength / 2))
-    val firstTurretInFirstLane: Plant = PeaShooter((1, LanesLength /2 - 1))
-    val secondTurretInSecondLane: Plant = PeaShooter((2, LanesLength * 0.75))
-    val firstZombieFirstLane: Enemy = Zombie((1, LanesLength))
-    val thirdZombieInSecondLane: Enemy = Zombie((2, LanesLength))
-    val entities: List[Entity] = List(firstTurretInFirstLane, secondTurretInSecondLane,
-      firstZombieFirstLane, thirdZombieInSecondLane)
-    assert(entities.filter(basicZombie.isInterestedIn) == List(firstTurretInFirstLane))
+  private val testingLane = 1
+  private val otherLane = 2
+  private val zombie: Troop = Troops.ofType[Zombie] withPosition (testingLane, 80)
+  private val fastZombie: Troop = Troops.ofType[FastZombie] withPosition (testingLane, 80)
+  private val warriorZombie: Troop = Troops.ofType[WarriorZombie] withPosition (testingLane, 80)
+
+  private val plantInOtherLane: Troop = Troops.ofType[PeaShooter] withPosition (otherLane, 10)
+  private val plantInTheSameLane: Troop = Troops.ofType[PeaShooter] withPosition (testingLane, 10)
+  private val plantInRange: Troop = Troops.ofType[PeaShooter] withPosition (testingLane, 75)
+
+  private val dummyPlant: Troop = Troops.ofType[PeaShooter] withPosition (otherLane, 50)
+  private val dummyZombie: Troop = Troops.ofType[Zombie] withPosition (otherLane, 50)
+  private val dummyBullet: Bullet = PeaBullet(0,0)
+
+
+  "A Zombie" should "be in Moving state if it can't attack any plant" in {
```

Comment:
More regarding the names: by describing the test as "A zombie should..." it seems that you're testing a generic Zombie.